lib/efi.py

```diff
@@ -309,7 +309,15 @@ def __init__(self, name, is_null=False):
         self.auth = os.path.join(self.tempdir.name, '%s.auth' % self.name)
 
     def is_signed(self):
-        return os.path.exists(self.auth)
+        return self._auth_data is not None
+
+    def auth_data(self):
+        assert self.is_signed()
+        return self._auth_data
+
+    def auth(self):
+        assert self.is_signed()
+        return self._auth
```

Comment:
I can't comment on the commit message, so I'm commenting here. 

The commit message says "For checking reasons, we don't want users to grab these variables
directly from the object."

I was not sure initially what this meant. After reading the code I assume this is so that we can run `assert self.is_signed()` every time we access them. Could you enrich the commit message with a description of the problem it solves?